babel/core.py

```diff
@@ -1112,26 +1150,32 @@ def negotiate_locale(preferred: Iterable[str], available: Iterable[str], sep: st
     return None
 
 
-def parse_locale(identifier: str, sep: str = '_') -> tuple[str, str | None, str | None, str | None]:
+def parse_locale(identifier: str, sep: str = '_') \
+        -> tuple[str, str | None, str | None, str | None, str | None]:
     """Parse a locale identifier into a tuple of the form ``(language,
-    territory, script, variant)``.
+    territory, script, variant, modifier)``.
 
     >>> parse_locale('zh_CN')
-    ('zh', 'CN', None, None)
+    ('zh', 'CN', None, None, None)
     >>> parse_locale('zh_Hans_CN')
-    ('zh', 'CN', 'Hans', None)
+    ('zh', 'CN', 'Hans', None, None)
     >>> parse_locale('ca_es_valencia')
-    ('ca', 'ES', None, 'VALENCIA')
+    ('ca', 'ES', None, 'VALENCIA', None)
     >>> parse_locale('en_150')
-    ('en', '150', None, None)
+    ('en', '150', None, None, None)
     >>> parse_locale('en_us_posix')
-    ('en', 'US', None, 'POSIX')
+    ('en', 'US', None, 'POSIX', None)
+    >>> parse_locale('it_IT@euro')
+    ('it', 'IT', None, None, 'euro')
+    >>> parse_locale('it_IT@custom')
+    ('it', 'IT', None, None, 'custom')
```

Comment:
I'd like to be sure this parses an empty modifier as None:
```suggestion
    >>> parse_locale('it_IT@custom')
    ('it', 'IT', None, None, 'custom')
    >>> parse_locale('it_IT@')
    ('it', 'IT', None, None, None)
```

Reply:
Makes sense! Thanks.